src/modules/gravityCompensator/main.cpp

```diff
@@ -227,12 +227,31 @@ class gravityModuleCompensator: public RFModule
 
         //------------SPECIAL PARAM TP DEFINE THE HEAD TYPE-----//
         version_tag icub_type;
-        if (rf.check("headV2"))
+        // set icub_type.head_version int to 1 (default) or 2
+        if (rf.check("head_version"))
         {
-            yInfo("'headV2' option found. Using icubV2 head kinematics.\n");
-            icub_type.head_version = 2;
+            if (rf.find("head_version").asDouble()==1.0)
+            {
+                yInfo("\"head_version\" %g requested => Using V1 head kinematics.\n",
+                      rf.find("head_version").asDouble());
+            }
+            else if (rf.find("head_version").asDouble()==2.0 ||
+                rf.find("head_version").asDouble()==2.5)
+            {
+                yInfo("\"head_version\" %g requested => Using V2 head kinematics.\n",
+                      rf.find("head_version").asDouble());
+                icub_type.head_version = 2;
```

Comment:
Hi @pattacini, the thing is that the struct [iCub::iDyn::version_tag](https://github.com/robotology/icub-main/blob/master/src/libraries/iDyn/include/iCub/iDyn/iDynBody.h#L108-L118) has a head version field which is only an integer, so how can I instantiate the correct 2.0 vs 2.5 kinematics?

Presently, `gravityCompensator` and `wholeBodyDynamics` are setting that head version integer to 1 (default) or 2. In particular, I had a look at `gravityCompensator` and it seems to me that it is not yet aware of any difference between head 2.0 vs 2.5 (please correct me if I'm wrong).

Basically, this is a small/incremental/cosmetic PR, only to preserve the current behaviour while aligning the syntax of the two modules in the title with the new recent syntax of `iKinGazeCtrl`, and have consistent parameter names in the `iCubStartup.xml` script for the end user.

By purpose, I ignored the deeper changes and implications between 2.0 and 2.5 head mentioned in https://github.com/robotology/icub-main/issues/285.